src/iono.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Dmitry Tatarinov <dmitry.tatarinov@exafore.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <assert.h>
+#include <string.h>
+
+#include "iono.h"
+#include "signal.h"
+
+static ionosphere_t iono_params[NUM_TRACKER_CHANNELS];
+static u32 iono_params_decoded_flag = 0;
+
+/** Stores ionospheric parameters
+ * \param ch_id Tracking channel ID
+ * \param params pointer to ionospheric parameters to be stored
+ */
+void gps_iono_params_store(u8 ch_id, ionosphere_t *params)
+{
+  assert(ch_id < NUM_TRACKER_CHANNELS);
+  assert(params != NULL);
+  memcpy(&iono_params[ch_id], params, sizeof(ionosphere_t));
```

Comment:
I think this will need a mutex to prevent store and read functions being called at same time.